### DISCUSSION TITLE
Providing Schema.Json

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5246] [Q&A] [answered]
In my Graph QL server (deployed as an Azure Function), I have provided a GET operation that will return the SDL of the services offered.

However, one of our consumer project team want this as a SCHEMA.JSON .. Is there an easy way to expose this?

**Comment by PascalSenn** [comment]
Do you mean the introspection result?

**Reply by bo55vxr** [reply]
Yes ... I have this code which produces as SDL ... But the consuming team want it as Json. They have tried using Apollo Sandbox, but we have CORS policies in place which prevent it working so I would like to be able to generate direct from an end-point

```
var executor = await _resolver.GetRequestExecutorAsync();
var response = req.CreateResponse(System.Net.HttpStatusCode.OK);
response.Headers.Add("Content-Type", "application/text");
response.WriteString(executor.Schema.ToString(), System.Text.Encoding.UTF8);
return response;
```

**Reply by michaelstaib** [reply]
Did you know that you can just call `http://yourserver/graphql?sdl` to fetch the sdl.

**Reply by michaelstaib** [reply]
To get the json representation you need to do an introspection request. you can create a persisted query for this if you want to provide kind of a rest endpoint for it. In this case the request would look like `http://yourserver/graphql?id=persistedQueryName`

**Reply by bo55vxr** [reply]
> Did you know that you can just call `http://yourserver/graphql?sdl` to fetch the sdl.

We have deployed as an Azure Function App (sorry, I should have mentioned that) ... I think the `?sdl` would only be applicable to Web App.

**Comment by bo55vxr** [comment, marked as answer]
Well, the simple answer was to get our Cloud Platform Team to resolve the CORS issue for Apollo Sandbox.
